Patient Registration Page:

Ananya-Naik-kinwits opened the pull request:
Details page
validation schema
success page all three added

Ananya-Naik-kinwits committed:
Paitent Registration Page:
Details page
validation schema
success page all three added

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+        </div>
+
+        {/* Tabs */}
+        <div className="relative mb-4 border-b border-gray-200">
```
why you are not using daisyUI tab classname here?

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                  {...inputProps}
+                />
+                <div className="flex gap-4">
+                  <div className="w-1/2">
```
instead of w-1/2, make it to `w-3/5`

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                      {...inputProps}
+                    />
+                  </div>
+                  <div className="w-1/2">
```
same here, `w-2/5`

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                      label="Blood Group"
+                      placeholder="Select"
+                      options={[
+                        { value: "A+", label: "A+" },
```
extract these options to an array of objects on top

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                      label="Gender"
+                      placeholder="Select Gender"
+                      options={[
+                        { label: "Male", value: "male" },
```
extract these options to an array of objects on top

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                <Textarea
+                  name="address"
+                  label="Address"
+                  placeholder="Full Address"
```
for placeholders `sentence case` is preferred.
change text to - `Enter full address`.

change for all placeholders where every applicable.

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+                  {...inputProps}
+                />
+                <Textarea
+                  name="currentillness"
```
maintain camelCasing for name.

Ex: `currentIllness`

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+          <Button
+            text="Complete"
+            className="btn-primary btn-lg flex-1"
+            onClick={async () => {
```
Scenario:

User filled all the required details in `personal` tab and he haven't switched his tab to `medical` tab, and `medical` tab has required fields, add a functionality to change tab to medical if he enters all required details in personal.

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/PatientDetails.jsx:
```diff
@@ -0,0 +1,367 @@
+            )}
+
+            {/* Personal Tab Fields */}
+            {activeTab === "personal" && (
```
Extract `Personal tab` fields to one component and `Medical tab` fields to another component and render conditionally.

Ex: {activeTab === "personal" ? `<PersonalTab />` : `<MedicalTab />`}

SaiPraneeth-Kinwits reviewed (changes requested):
Kindly resolve the requested changes.

SaiPraneeth-Kinwits commented on src/pages/auth/RegisterPatient/RegisterPatientSchema.js:
```diff
@@ -0,0 +1,52 @@
+    }),
+});
+
+export default RegisterPatientSchema;
```
you already did export in line: 11, why again `default` export here?

Ananya-Naik-kinwits committed:
Patient Details Page:
implemented using daisyUI
and added a temporary compulsory validation for "currentMedication" in schema

Ananya-Naik-kinwits committed:
PRP final changes:
validation and tabs

SaiPraneeth-Kinwits reviewed (approved):
Approved!
